### PR TITLE
use consistent naming for custom gateways

### PR DESCRIPTION
**Pull request by ayj**

### CODE REVIEW

File: manifests/gateways/istio-ingress/templates/gateway.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{ $gateway := index .Values "gateways" "istio-ingressgateway" }}
+  {{- if eq $gateway.name "istio-ingressgateway" }}
```
Comment (howardjohn) [inline manifests/gateways/istio-ingress/templates/gateway.yaml]: I think this file is gone now

File: manifests/gateways/istio-ingress/templates/sidecar.yaml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+{{- $gateway := index .Values "gateways" "istio-ingressgateway" }}
+{{- if eq $gateway.name "istio-ingressgateway" }}
```
Comment (howardjohn) [inline manifests/gateways/istio-ingress/templates/sidecar.yaml]: this should be gone now as well

**Review by howardjohn — COMMENTED** [review]
PBD change lgtm

**Comment by ayj** [comment]
Deleted files have been removed.

File: operator/cmd/mesh/testdata/manifest-generate/output/gateways_override_default.golden.yaml [hunk 3]
```diff
@@ -797,7 +797,7 @@ spec:
 apiVersion: policy/v1beta1
 kind: PodDisruptionBudget
 metadata:
-  name: ingressgateway
+  name: istio-ingress-1
```
Comment (elfinhe) [inline operator/cmd/mesh/testdata/manifest-generate/output/gateways_override_default.golden.yaml]: Why does this end with “-1”

Comment (ayj) [reply operator/cmd/mesh/testdata/manifest-generate/output/gateways_override_default.golden.yaml]: That's the gateway name in the input gateways_override_default.yaml

Comment (elfinhe) [reply operator/cmd/mesh/testdata/manifest-generate/output/gateways_override_default.golden.yaml]: Thanks.

**Review by elfinhe — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Pull request merged**